File: MolenApplicatie.Server/Models/MolensResponseType.cs
namespace MolenApplicatie.Server.Models
{
    public class MolensResponseType
    {
        public int ActiveMolensWithImage { get; set; }
        public int RemainderMolensWithImage { get; set; }
        public int TotalMolensWithImage { get; set; }
        public int TotalCountActiveMolens { get; set; }
        public int TotalCountRemainderMolens { get; set; }
        public List<CountDisappearedMolens> TotalCountDisappearedMolens { get; set; }
        public int TotalCountExistingMolens { get; set; }
        public int TotalCountMolens { get; set; }
        public List<MolenData> Molens { get; set; }
    }

    public class CountDisappearedMolens
    {
        public string Provincie { get; set; }
        public int Count { get; set; }
    }
}

File: molenapplicatie.client/src/Interfaces/MolensResponseType.ts
import { MolenData, MolenImage } from '../api/generated/data-contracts';

export interface MolensResponseType<T> {
  activeMolensWithImage: number;
  remainderMolensWithImage: number;
  totalMolensWithImage: number;
  totalCountActiveMolens: number;
  totalCountRemainderMolens: number;
  totalCountDisappearedMolens: CountDisappearedMolens[];
  totalCountExistingMolens: number;
  totalCountMolens: number;
  molens: T[];
  recentAddedImages?: RecentAddedImages[];
}

export interface CountDisappearedMolens {
  provincie: string;
  count: number;
}

export interface RecentAddedImages {
  molen: MolenData;
  images: MolenImage[];
}
